app/models/base_input.rb

```diff
@@ -0,0 +1,39 @@
+class BaseInput
```

Comment:
How would you feel about this being a module – included in `Input` and `InitializerInput` – rather than a base class? It seems like both classes need to share some implementation (hence a module) but [cannot be substituted](https://en.wikipedia.org/wiki/Liskov_substitution_principle) for one another.